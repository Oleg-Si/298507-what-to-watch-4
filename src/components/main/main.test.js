import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main.jsx';

const promoFilmMock = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`
};

const films = [
  {
    id: 0,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`
  },
  {
    id: 2,
    title: `Macbeth`,
    img: `img/macbeth.jpg`
  },
  {
    id: 3,
    title: `Aviator`,
    img: `img/aviator.jpg`
  }
];

const onFilmCardTitleClick = () => {};

it(`Main should render main screen`, () => {
  const tree = renderer.create(
      <Main
        promoFilmMock={promoFilmMock}
        films={films}
        onFilmCardTitleClick={onFilmCardTitleClick}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
