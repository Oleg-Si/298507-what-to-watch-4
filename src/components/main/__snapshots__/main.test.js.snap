// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяет снепшот компонента Main 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
                <symbol
                  id="play-s"
                  viewBox="0 0 19 19"
                >
                  <path
                    clipRule="evenodd"
                    d="M0 0L19 9.5L0 19V0Z"
                    fill="#EEE5B5"
                    fillRule="evenodd"
                  />
                </symbol>
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentary
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="preloader"
            style={
              Object {
                "left": "50%",
                "marginLeft": "-32px",
                "marginTop": "-32px",
                "position": "absolute",
                "top": "50%",
                "zIndex": "2",
              }
            }
          >
            <svg
              height="64px"
              viewBox="0 0 128 128"
              width="64px"
              xmlSpace="preserve"
            >
              <rect
                fill="#FFFFFF"
                height="100%"
                width="100%"
                x="0"
                y="0"
              />
              <g>
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="16"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="14.344"
                  transform="rotate(45 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="12.531"
                  transform="rotate(90 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="10.75"
                  transform="rotate(135 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="10.063"
                  transform="rotate(180 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="8.063"
                  transform="rotate(225 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="6.438"
                  transform="rotate(270 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="5.375"
                  transform="rotate(315 64 64)"
                />
                <animateTransform
                  attributeName="transform"
                  calcMode="discrete"
                  dur="720ms"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
                />
              </g>
            </svg>
          </div>
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              poster="img/aviator.jpg"
              src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/0"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="preloader"
            style={
              Object {
                "left": "50%",
                "marginLeft": "-32px",
                "marginTop": "-32px",
                "position": "absolute",
                "top": "50%",
                "zIndex": "2",
              }
            }
          >
            <svg
              height="64px"
              viewBox="0 0 128 128"
              width="64px"
              xmlSpace="preserve"
            >
              <rect
                fill="#FFFFFF"
                height="100%"
                width="100%"
                x="0"
                y="0"
              />
              <g>
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="16"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="14.344"
                  transform="rotate(45 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="12.531"
                  transform="rotate(90 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="10.75"
                  transform="rotate(135 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="10.063"
                  transform="rotate(180 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="8.063"
                  transform="rotate(225 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="6.438"
                  transform="rotate(270 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="5.375"
                  transform="rotate(315 64 64)"
                />
                <animateTransform
                  attributeName="transform"
                  calcMode="discrete"
                  dur="720ms"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
                />
              </g>
            </svg>
          </div>
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              poster="img/macbeth.jpg"
              src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="preloader"
            style={
              Object {
                "left": "50%",
                "marginLeft": "-32px",
                "marginTop": "-32px",
                "position": "absolute",
                "top": "50%",
                "zIndex": "2",
              }
            }
          >
            <svg
              height="64px"
              viewBox="0 0 128 128"
              width="64px"
              xmlSpace="preserve"
            >
              <rect
                fill="#FFFFFF"
                height="100%"
                width="100%"
                x="0"
                y="0"
              />
              <g>
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="16"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="14.344"
                  transform="rotate(45 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="12.531"
                  transform="rotate(90 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="10.75"
                  transform="rotate(135 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="10.063"
                  transform="rotate(180 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="8.063"
                  transform="rotate(225 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="6.438"
                  transform="rotate(270 64 64)"
                />
                <circle
                  cx="16"
                  cy="64"
                  fill="#000000"
                  fillOpacity="1"
                  r="5.375"
                  transform="rotate(315 64 64)"
                />
                <animateTransform
                  attributeName="transform"
                  calcMode="discrete"
                  dur="720ms"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
                />
              </g>
            </svg>
          </div>
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              poster="img/snatch.jpg"
              src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Snatch
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
